rest/src/test/java/org/envirocar/server/rest/util/OSMTileRendererTests.java

```diff
@@ -29,4 +34,65 @@ public void SaveImageWorks() throws IOException {
 			osmt.saveImage(new BufferedImage(768, 512, BufferedImage.TYPE_INT_RGB), "1235456");
 	}
 	
+	@Test
+	public void zoomLevelTest(){
+		int level=osmt.getZoomLevel(populateCoordinates());
+		assertTrue("Zoom level exceeds limit", 19 >= level);
+		assertTrue("Zoom level lower than limit",  0  <= level);
+	}
+	
+	@Test
+	public void colorCodeTest(){
+		Random r = new Random(); 
+		Color color=osmt.getColorCode(r.nextDouble() * 90.0);
+		assertTrue(color.getRGB() != 0);
+	}
+	
+	@Test
+	public void tileDetailTest() {
+		double lat = 0.0; 
+	    double lon = 0.0; 
+	    int zoom = 19;
+		int [] tileDetails=osmt.getTileDetails(lon,lat,zoom);
+		assertEquals (tileDetails[0],262144);
+		assertEquals (tileDetails[1],262144);
+	}
+	
+	@Test
+	public void validateTile(){
+		Random r = new Random(); 
+		double lat = r.nextDouble() * 90.0; 
+	    double lon = r.nextDouble() * 180.0; 
+	    int zoom = r.nextInt(19);
+		int [] tileDetails=osmt.getTileDetails(lon,lat,zoom);
+		assertTrue(tileDetails[0] <= Math.pow(2, 2*zoom));
+		assertTrue(tileDetails[1] <= Math.pow(2, 2*zoom));
```

Comment:
See above. It is tested if the general range of the details is matched. Use code-defined coordinates and zoomLevel and use your knowledge of what the result should be.